doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -220,63 +220,79 @@ Requests to the environment
-   statement contains :n:`@qualid`. This command is useful to remind the user
-   of the name of library lemmas.
 
-   .. exn:: The reference @qualid was not found in the current environment.
+.. cmd:: Search {+ @search_item } {? {| inside | outside } {+ @qualid } }
 
-      There is no constant in the environment named qualid.
+   .. insertprodn search_item search_item
+
+   .. prodn::
+      search_item ::= {? - } @string {? % @ident }
+      | {? - } @one_term
+
+   This command displays the name and type of objects (hypothesis of
+   the current goal, theorems, axioms, etc) in the current context.
+   This command is useful to remind the user of the names of library lemmas.
 
-   .. cmdv:: Search @string
+   .. todo above: "current context" meaning on the current goal or the first goal?
+      below: adapted the old wording, doesn't seem consistent
+      also will "invocation-of-tactics" cover the command case, or will there be a new section?
 
-      If :n:`@string` is a valid identifier, this command
-      displays the name and type of all objects (theorems, axioms, etc) of
-      the current context whose name contains string. If string is a
-      notation’s string denoting some reference :n:`@qualid` (referred to by its
-      main symbol as in `"+"` or by its notation’s string as in `"_ + _"` or
-      `"_ 'U' _"`, see Section :ref:`notations`), the command works like ``Search`` :n:`@qualid`.
+   * :n:`@string  {? % @ident }` - If :n:`@string` is a valid identifier,
+     search for objects whose name contains :n:`@string`. If :n:`@string` is a
+     notation's string for some reference :n:`@qualid` (referred to by its
+     main symbol as in `"+"` or by its notation’s string as in `"_ + _"` or
+     `"_ 'U' _"`, see Section :ref:`notations`) the command works like :cmd:`Search` :n:`@qualid`.
 
-   .. cmdv:: Search @string%@ident
+     .. todo: don't understand the 2nd sentence; notation's string -> notation string?
+        what if it's not a valid identifier?  error or does something different?
+        this is a substring match, e.g. "abc" matches "bc", right?
 
-      The string string must be a notation or the main
-      symbol of a notation which is then interpreted in the scope bound to
-      the delimiting key :token:`ident` (see Section :ref:`LocalInterpretationRulesForNotations`).
+     .. todo would be nice if :n: had syntax for embedding :cmd: et al
 
-   .. cmdv:: Search @term_pattern
+     Specifying :n:`@ident` limits this to interpretation in the scope bound to
+     the delimiting key :token:`ident` (see Section :ref:`LocalInterpretationRulesForNotations`).
 
-      This searches for all statements or types of
-      definition that contains a subterm that matches the pattern
-      :token:`term_pattern` (holes of the pattern are either denoted by `_` or by
-      :n:`?@ident` when non linear patterns are expected).
+     .. todo: does delimiting key mean anything special?
```

Comment:
Usual meaning from the notation documentation.

Reply:
Took me maybe an hour to (sort of) figure what was meant here.  I've not used scopes at all before.  Not easy to figure out from the doc without reading a lot and fairly closely.

Reply:
Indeed, but I just meant that there is nothing special to explain here. That being said, the "Syntax extensions" chapter defines a `scope` non-terminal (that's just an `ident`). It could make sense to replace all the `% @ident` with `% @scope` so that there is a direct link to the documentation of scopes. Like we replaced a bunch of `: @term` by `: @type`.

Reply:
Will do.  "scope" is a bit problematic because the doc uses it many times for it seems multiple meanings, eg "interpretation scope", Ltac2 scope, I think in the description of sections and in the ordinary sense of the word.

Ex: "By default, the scope of :cmd:`Opaque` is limited to the current section or file."

Reply:
Right, but I think this is still the most common use case. We could try to reduce the other uses.